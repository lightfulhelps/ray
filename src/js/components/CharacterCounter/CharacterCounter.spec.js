import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';
import { CharacterCounter } from '../../';

const setup = (overrides = {}) => {
  const props = merge({}, overrides);
  const wrapper = shallow(<CharacterCounter {...props} />);

  return { wrapper, props };
};

describe('<CharacterCounter />', () => {
  it('should render', () => {
    const { wrapper } = setup({ text: 'Text', max: 5 });

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle className', () => {
    const { wrapper } = setup();

    wrapper.setProps({ className: 'custom' });

    expect(wrapper.hasClass('character-counter')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });

  it('should handle max', () => {
    const { wrapper } = setup({ max: 10 });

    expect(wrapper.text()).toEqual('<Icon />0/10');
  });

  it('should handle text', () => {
    const { wrapper } = setup({ text: 'This is my text' });

    expect(wrapper.text()).toEqual('<Icon />15');
  });

  it('should pass through other props', () => {
    const { wrapper } = setup({ tabIndex: 1, id: 'test' });

    expect(wrapper.prop('tabIndex')).toEqual(1);
    expect(wrapper.prop('id')).toEqual('test');
  });

  it('should add danger class if over max', () => {
    const { wrapper } = setup({ text: 'Foo', max: '10' });

    expect(wrapper.hasClass('text-danger')).toEqual(false);

    wrapper.setProps({ text: 'Some longer text' });

    expect(wrapper.hasClass('text-danger')).toEqual(true);
  });
});
