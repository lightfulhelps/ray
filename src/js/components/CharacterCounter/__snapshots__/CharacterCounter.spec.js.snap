// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterCounter /> should render 1`] = `
<div
  className="character-counter"
>
  <Icon
    name="multipleSocialProvider"
  />
  4
  /5
</div>
`;
